blog/_posts/2018-06-10-bsp.md

```diff
@@ -0,0 +1,214 @@
+overhead" would be too high.
+
+Instead it is much better if we, tooling developers, can bear the burden of
+shipping tools that integrate with your existing workflow and you, as a user,
+can just install them with a quick search and a single click.
+
+Today, I introduce the Build Server Protocol, a project that I believe plays
+a fundamental role in bringing us closer to make this vision a reality.
+
+## Any IDE, Any Build Tool
+
+Traditionally, every IDE has implemented a custom integration for each
+supported build tool in order to extract build information such as classpaths
+or source directories for every project in your build. This information is
+necessary for the IDE to present you with a workspace where you can browse,
+write and compile Scala code. This process takes place every time you "Import
+Project" from the IDE user interface (think IntelliJ or Eclipse).
+
+Likewise, all build tools are expected to integrate with all available IDEs.
+Most of them provide "Export Project" capabilities so that user adoption is
+not hurt by lack of official IDE support. This support isn't as thorough and
+polished as the one provided by IDEs, but usually gets the job done at the
+cost of supported features.
+
+The growing number of language servers and build tools in the wider
+programming community means tooling developers spend a lot of time working on
+these integrations, as this work needs to be duplicated on each side every
+time there is a new IDE or build tool. This is especially true in the case of
+Scala where there is a lot of activity in the build tools space and new build
+tools such as [Bazel], [Pants] and [mill] have started to gain adoption among
+Scala developers, making it hard to support them all.
+
+The idea behind the Build Server Protocol is to standardize the protocol for
+how build tools (servers) and IDEs (clients) communicate, so a single Build
+Server can be re-used in multiple IDEs, and IDEs can support build tools with
+minimal effort.
+
+As a result, end users are provided with the best developer experience and
+tooling developers can improve the quality of their integrations with less
+effort and time, without locking out a part of the Community from good Scala
+language support.
+
+### Relationship with the Language Server Protocol
+
+If you have heard of Microsoft's [Language Server Protocol][LSP], you may be
+wondering what's the relationship between these two protocols. BSP and LSP
+are complementary by design.
+
+While LSP specifies endpoints for communication between an *editor acting as
+a client* and a language server, BSP specifies endpoints between a *language
+server acting as a client* and a build server.
+
+It turns out that if we want to support Scala across editors like Visual
+Studio Code, `vim` or Sublime via a Scala language server, we also need to
+integrate with build tools to support features like compiler diagnostics or
+code navigation. One of the main focus of BSP is that it can be implemented
```

Comment:
One of the main foci*

Reply:
Euax ad Latinam!

Reply:
oh wow, i feel so cultivated using this word :smile:

Reply:
OK guys, i got impostor syndrome and decided to go for the classic "properties".